src/Functions/FunctionCustomWeekToSomething.h

```diff
@@ -45,6 +50,23 @@ class FunctionCustomWeekToSomething : public IFunctionCustomWeek<Transform>
                 arguments[0].type->getName(), this->getName());
     }
 
+    std::pair<Field, Field> convertStringToUnixTimestamp(const Field & left, const Field & right) const override
+    {
+        ColumnsWithTypeAndName temp_block;
+        auto src_type = std::make_shared<DataTypeString>();
+
+        auto column = src_type->createColumn();
+        column->insert(left);
+        column->insert(right);
+        temp_block.emplace_back(ColumnWithTypeAndName{std::move(column), src_type,"tmp"});
+
+        auto to_datetime = FunctionFactory::instance().get("toUnixTimestamp", context);
+        auto res_column = to_datetime->build(temp_block)->execute(
```

Comment:
Why not just call `readDateTimeText` src/IO/ReadHelpers.h?

Reply:
@Enmk This `SELECT toDayOfWeek(toString(toUnixTimestamp(now())))` works well and `readDateTimeText` cannot process the timestamp, right?

Reply:
It can't process it because it requires parsing a string, and this is exactly what you are trying to fix. Im just curious why would you need to instantiate a function, create input columns, extract values from output column individually when you can call a function here and do all that 10 times more concise direct and simpler.

But perhaps I am missing some crucial point here... could you please elaborate why you are calling a funtion instead?

Reply:
Ok, Here we need to find a universal way to process all data type in the string, like Date, DateTime, DateTime64, Timestamp, and I just referred to an older version implementation of `FunctionCustomWeekToSomething::executeImpl()`, it use a function to convert and I follow it, using `toUnixTimstamp` to cover all data types. Now it seems like things have changed, and I'll try to find a new way if possible.

Reply:
@Enmk Hello, I just found 2 things: 
1. `toDateTime64` have different behavior between timestamp of string type and integer type if number is large, and then they'll cause different result when passing to `toDayOfWeek`:
```
SELECT
    toDateTime64(10413763199, 3) AS largest_datetime64,
    toDateTime64('10413763199', 3) AS str_largest_datetime64,
    toDateTime64(20413763199, 3) AS strange_overflow_datetime64,
    toDateTime64('20413766799', 3) AS str_strange_overflow_datetime64


Row 1:
──────
largest_datetime64:              2299-12-31 23:59:59.000
str_largest_datetime64:          2299-12-31 23:59:59.000
strange_overflow_datetime64:     2299-12-31 23:59:59.000
str_strange_overflow_datetime64: 1970-08-25 14:29:26.799
```

2. `toDayOfWeek` cannot process a 'timestamp' of string type larger than the max value:
```
SELECT toDayOfWeek('10413763199')

Received exception from server (version 23.12.1):
Code: 41. DB::Exception: Received from localhost:9000. DB::ParsingException. DB::ParsingException: Cannot read DateTime: unexpected number of decimal digits: 11: While processing toDayOfWeek('10413763199'). (CANNOT_PARSE_DATETIME)

SELECT toDayOfWeek(toString(toUInt32(-1)))

Row 1:
──────
toDayOfWeek(toString(toUInt32(-1))): 7
```

So I think we don't need to consider when datetime or timestamp overflows in `getMonotonicityForRange` because the undefined behavior, and It may be a better way to convert field data to `DateTime64` type to make it compatible to as many types as possible. Do you have any good ideas? To be honest, I'm not very familiar with time conversion.